text/1060-tracked-promise.md

```diff
@@ -0,0 +1,494 @@
+
+[gh-release-plan]: https://github.com/embroider-build/release-plan
+[gh-emberjs]: https://github.com/emberjs/ember.js/
+
+Why `type=module`?
+
+This is a requirement for some optimization features of packages (webpack / vite), such as _proper_ treeshaking -- without `type=module`, the best optimization we can get is "pay for only what you import". For large projects this isn't so much of a problem, but for small projects (or highly optimized projects), the impact to network transfer/parse/eval is measurable. This RFC is also proposing that `@ember/reactive` be _the_ place for all our ecosystem's reactivity utilities will end up once they've been proven out, tested, and desire for standardation is seen.
+
+For example, other future exports from `@ember/reactive` (in future RFCs), may include:
+- TrackedObject
+- TrackedArray
+- TrackedMap
+- TrackedSet
+- TrackedWeakSet
+- TrackedWeakMap
+- localCopy
+- certain [window properties](https://svelte.dev/docs/svelte/svelte-reactivity-window)
+- ...and more
+
+without the static analysis guarantees of `type=module`, every consumer of `@ember/reactive` would always have all of these exports in their build.
+For some utilities, we can place them under sub-path-exports, such as `@ember/reactive/window`, for window-specific reactive properties, but the exact specifics of each of these can be hashed out in their individual RFCs.
+
+
+### Consumption
+
+When a project wants to use `@ember/reactive`, they would then only need to install the package separately / add it to their `package.json`.
+
+The proposed list of compatibilyt here is only meant as an example -- if implementation proves that more can be supported easier, with less work, that should be pursued, and this part is kind of implementation detail.
+
+But for demonstration:
+- apps pre [version available], would add `@ember/reactive` to their `devDependencies` or `dependencies`
+  - importing `@ember/reactive` would be handled by ember-auto-import/embroider (as is the case with all v2 addons)
+- v1 addons would not be supported
+- v2 addons, for maximum compatibility, would need to add `@ember/reactive` to their `dependencies`
+  - in consuming apps post [version available], this would be optimized away if the version declared in dependencies satisfies the range provided by the consuming app (an optimization that packagers already do, and nothing we need to worry about)
+- apps post [version available], would not need to add `@ember/reactive` to their `devDependencies` or `dependencies`, as we can rely on the `ember-addon#renamed-modules` config in ember-source's `package.json`.
+
+## How we teach this
+
+### API Docs
+
+#### `trackPromise`
```

Comment:
`trackedPromise` feels better than `trackPromise`, since the former communicates the type of thing generated and the latter indicates an action, and quite often `trackPromise` implementations exist which do in fact track promises but not to make them reactive but rather for metrics and reporting or timing etc.

Reply:
Renamed! Appreciate the explanation!

And noun vs action is an important distinction!